crates/astria-conductor/tests/blackbox/soft_and_firm.rs

```diff
@@ -87,20 +124,135 @@ async fn simple() {
+///    state update of the current height happen before receipt of the next block.
+/// 3. Mounts are made for firm and soft update commitment state calls, with the soft mount
+///    expecting exactly 0 calls.
+/// 4. 1000ms is allotted for the conductor to execute the block and update the firm commitment
+///    state, noting that this allows time to test for an erroneously updated soft commitment state
+///    before the conductor receives the next block.
+/// 5. 2000ms is allotted for the conductor to execute the next block and update the soft commitment
+///    state at the next height.
+#[allow(clippy::too_many_lines)] // allow: all mounts and test logic are necessary.
+#[tokio::test(flavor = "multi_thread", worker_threads = 1)]
+async fn executes_firm_then_soft_at_next_height() {
+    let test_conductor = spawn_conductor(CommitLevel::SoftAndFirm).await;
+
+    mount_get_genesis_info!(
+        test_conductor,
+        sequencer_genesis_block_height: 1,
+        celestia_block_variance: 10,
+    );
+
+    mount_get_commitment_state!(
+        test_conductor,
+        firm: (
+            number: 1,
+            hash: [1; 64],
+            parent: [0; 64],
+        ),
+        soft: (
+            number: 1,
+            hash: [1; 64],
+            parent: [0; 64],
+        ),
+        base_celestia_height: 1,
     );
 
+    mount_abci_info!(
+        test_conductor,
+        latest_sequencer_height: 4,
+    );
+
+    mount_sequencer_genesis!(test_conductor);
+
+    mount_celestia_header_network_head!(
+        test_conductor,
+        height: 1u32,
+    );
+
+    mount_celestia_blobs!(
+        test_conductor,
+        celestia_height: 1,
+        sequencer_heights: [3],
+    );
+
+    mount_sequencer_commit!(
+        test_conductor,
+        height: 3u32,
+    );
+
+    mount_sequencer_validator_set!(test_conductor, height: 2u32);
+
     let execute_block = mount_executed_block!(
         test_conductor,
         number: 2,
         hash: [2; 64],
         parent: [1; 64],
     );
 
-    let update_commitment_state_soft = mount_update_commitment_state!(
+    // Mount soft block at current height with a slight delay
+    mount_get_filtered_sequencer_block!(
+        test_conductor,
+        sequencer_height: 3,
+        delay: Some(Duration::from_millis(500)),
+    );
+
+    // Mount soft block at next height with substantial delay
+    mount_get_filtered_sequencer_block!(
+        test_conductor,
+        sequencer_height: 4,
+        delay: Some(Duration::from_millis(1000)),
+    );
+
+    let update_commitment_state_firm = mount_update_commitment_state!(
         test_conductor,
+        firm: (
+            number: 2,
+            hash: [2; 64],
+            parent: [1; 64],
+        ),
+        soft: (
+            number: 2,
+            hash: [2; 64],
+            parent: [1; 64],
+        ),
+        base_celestia_height: 1,
+    );
+
+    let _stale_update_soft_commitment_state = mount_update_commitment_state!(
```

Comment:
This would be very surprising to a reader: make a comment above this to explain why this is bound like that and not `wait_until_satisfied` (answer: so that the mock condition is tested on drop; this should actually be mounted on the server instead of returning a guard, but is a workaround for how the helpers for the conductor blackbox tests are implemented).